doctr/datasets/synthtext.py

```diff
@@ -0,0 +1,103 @@
+# Copyright (C) 2021, Mindee.
+
+# This program is licensed under the Apache License version 2.
+# See LICENSE or go to <https://www.apache.org/licenses/LICENSE-2.0.txt> for full license details.
+
+import os
+from pathlib import Path
+from typing import Any, Callable, Dict, List, Optional, Tuple
+
+import numpy as np
+import scipy.io as sio
+from tqdm import tqdm
+
+from .datasets import VisionDataset
+
+__all__ = ['SynthText']
+
+
+class SynthText(VisionDataset):
+    """SynthText dataset from `"Synthetic Data for Text Localisation in Natural Images"
+    <https://arxiv.org/abs/1604.06646>`_.
+
+    Example::
+        >>> # NOTE: This dataset has currently no train/test split
+        >>> from doctr.datasets import SynthText
+        >>> data_set = SynthText(download=True)
+        >>> img, target = data_set[0]
+
+    Args:
+        sample_transforms: composable transformations that will be applied to each image
+        rotated_bbox: whether polygons should be considered as rotated bounding box (instead of straight ones)
+        **kwargs: keyword arguments from `VisionDataset`.
+    """
+
+    URL = 'https://thor.robots.ox.ac.uk/~vgg/data/scenetext/SynthText.zip'
+    SHA256 = '28ab030485ec8df3ed612c568dd71fb2793b9afbfa3a9d9c6e792aef33265bf1'
+
+    def __init__(
+        self,
+        train: bool = True,
```

Comment:
There is no train/val split?
If so, we should make it ourselves (say the first 90% goes into train, 10% for val)

Reply:
```
if train:
            self.data = self.data[:int(len(self.data) * 0.9)]
        else:
            self.data = self.data[int(len(self.data) * 0.9):]
```
@fg-mindee 
ok ?

Reply:
well let's avoid unnecessary iterations:
- if train --> at 90% we break the iteration
- if val --> we start iterating only at 90%